manifests/client/redhat.pp

```diff
@@ -1,58 +1,61 @@
 # Specific settings for client on Redhat distribution.
 class nfs::client::redhat inherits nfs::base {
 
-  package { 'nfs-utils':
-    ensure => present,
-  }
-
-  if versioncmp($::operatingsystemmajrelease, '6') == 0 {
-
-    package {'rpcbind':
-      ensure => present,
+  case  $::operatingsystemmajrelease {
+    '5' : {
+      $nfslock_requirement = [Package['nfs-client'], Package['nfs-utils']]
+      $nfsclient_package   = 'portmap'
+      $nfsclient_service   = 'portmap'
+      $nfslock_service     = 'nfslock'
+      $netfs_requirement  = [Service['nfs-client'], Service['nfslock']]
     }
-
-    service {'rpcbind':
-      ensure    => running,
-      enable    => true,
-      hasstatus => true,
-      require   => [Package['rpcbind'], Package['nfs-utils']],
+    '6' : {
+      $nfslock_requirement = [Service['nfs-client']]
+      $nfslock_service     = 'nfslock'
+      $nfsclient_service   = 'rpcbind'
+      $nfsclient_package   = 'rpcbind'
+      $netfs_requirement   = [Service['nfslock']]
     }
-
-  } else {
-
-    package { 'portmap':
-      ensure => present,
+    '7': {
+      $nfslock_requirement = [Service['nfs-client']]
+      $nfslock_service     = 'nfs-lock'
+      $nfsclient_service   = 'rpcbind'
+      $nfsclient_package   = 'rpcbind'
+      $netfs_requirement   = undef
     }
-
-    service { 'portmap':
-      ensure    => running,
-      enable    => true,
-      hasstatus => true,
-      require   => [Package['portmap'], Package['nfs-utils']],
+    default : {
+      fail('This Major release is not supported')
     }
-
   }
-
-  $nfslock_requirement = $::operatingsystemmajrelease ? {
-    '6'     => Service['rpcbind'],
-    default => [Package['portmap'], Package['nfs-utils']]
+  # common items for all versions
+  $nfsclient_requirement  = [Package['nfs-client'], Package['nfs-utils']]
+  package { 'nfs-utils':
+    ensure => present,
   }
 
-  service {'nfslock':
+  service { 'nfslock':
     ensure    => running,
+    name      => $nfslock_service,
     enable    => true,
     hasstatus => true,
     require   => $nfslock_requirement,
   }
-
-  $netfs_requirement = $::operatingsystemmajrelease ? {
-    '6'     => Service['nfslock'],
-    default => [Service['portmap'], Service['nfslock']],
+  if  versioncmp($::operatingsystemmajrelease, '7')  != 0 {
```

Comment:
@ckaenzig well this bloc will be executed on RedHat8 when release, you should use `versioncmp($::operatingsystemmajrelease, '7')  < 0` instead